oeps/oep-0017-feature-toggles.rst

```diff
@@ -0,0 +1,809 @@
+
++-----------------+--------------------------------------------------+
+| OEP             | :doc:`OEP-0017 </oeps/OEP-0017-feature-toggles>` |
++-----------------+--------------------------------------------------+
+| Title           | Feature Toggles                                  |
++-----------------+--------------------------------------------------+
+| Last Modified   | 2018-02-07                                       |
++-----------------+--------------------------------------------------+
+| Authors         | Nimisha Asthagiri <nimisha@edx.org>              |
++-----------------+--------------------------------------------------+
+| Arbiter         | Eric Fischer <efischer@edx.org>                  |
++-----------------+--------------------------------------------------+
+| Status          | Under Review                                     |
++-----------------+--------------------------------------------------+
+| Type            | Best Practice                                    |
++-----------------+--------------------------------------------------+
+| Created         | 2018-02-14                                       |
++-----------------+--------------------------------------------------+
+| `Review Period` | 2018-02-14 - 2018-03-07                          |
++-----------------+--------------------------------------------------+
+| `Resolution`    |                                                  |
++-----------------+--------------------------------------------------+
+
+.. contents::
+   :local:
+   :depth: 2
+
+Abstract
+========
+
+A feature toggle is a software development technique that decouples 
+deployment of code from release (enablement) of the code. When used with care, 
+it can be a powerful tool in a continuous deployment environment to deploy 
+changes incrementally with the main codebase while under development. 
+Additionally, it allows a team to reduce the risk of breaking 
+production systems by providing levers to progressively test changes and quickly 
+disable changes if needed. For a large platform, it is also used to 
+selectively enable a change/feature for certain users or certain deployments.
+
+Feature toggles inherently add complexity to the code, however, since they 
+introduce multiple paths and configuration operations, with the potential to
+create an explosion in testing permutations and to create forgotten latent unused
+code paths. A thorough understanding of their implications and best practices is
+warranted to use them in a large-scale system.
+
+Motivation
+==========
+
+Given the many benefits of using feature toggles, edX development teams have
+been using them since the very start. Additionally, since deployment latency
+for edx.org has reduced from monthly->weekly->daily and since teams have been
+bitten too many times by `long-term feature branches`_, teams now have a greater 
+incentive to develop incrementally within the main branch. Updating veteran 
```

Comment:
I'm a bit confused by the use of "veteran" here. Is it referring to legacy code/UX?